File: lib/schema.ts
type Method = "POST" | "GET" | "DELETE" | "PUT" | "PATCH";

export type ApiType =
  | "string"
  | "int32"
  | "int64"
  | "bool"
  | "float"
  | "map"
  | "any"
  | "stringArray"
  | "int32Array"
  | "int64Array"
  | "anyArray"
  | "boolArray"
  | "floatArray"
  | "mapArray";

export interface Field {
  key: string;
  description?: string;
  type?: ApiType;
  typeOfObject?: Field[];
  typeOfObjectArray?: Field[];
  optional?: boolean;
  regex?: string;
  enum?: string[];
  min?: number;
  max?: number;
}

export const schemaData: Schema[] = [];

interface Schema {
  description?: string;
  url: string;
  method: Method;
  input: Field[];
  output: Field[];
}

export function schema(data: Schema) {
  schemaData.push(data);
}

File: old-golang/apigenerate/schema.go
package apigenerate

type ApiType string

const (
	String      ApiType = "string"
	Int         ApiType = "int32"
	BigInt      ApiType = "int64"
	Bool        ApiType = "bool"
	Float       ApiType = "float"
	Map         ApiType = "map"
	Any         ApiType = "any"
	StringArray ApiType = "[]string"
	IntArray    ApiType = "[]int"
	AnyArray    ApiType = "[]any"
	BoolArray   ApiType = "[]any"
	FloatArray  ApiType = "[]float"
	MapArray    ApiType = "[]map"
)

type Method string

const (
	GET    Method = "GET"
	POST   Method = "POST"
	PUT    Method = "PUT"
	PATCH  Method = "PATCH"
	DELETE Method = "DELETE"
)

type Param struct {
	Name              string
	Description       string
	TypeOfObject      []Param
	TypeOfArrayObject []Param
	Type              ApiType
	Optional          bool
	Min               int
	Max               int
	Regex             string
	Pick              []string
}

type Schema struct {
	Description string
	Url         string
	Method      Method
	Input       []Param
	Output      []Param
}

var schemaData []Schema = []Schema{}

func AddSchema(schema Schema) {
	schemaData = append(schemaData, schema)
}
